tool/lib/commands/serve.dart

```diff
@@ -271,23 +272,34 @@ class ServeCommand extends Command {
     }
 
     // This call will not exit until explicitly terminated by the user.
+    final cliCommand = CliCommand.dart([
+      if (debugServer) ...['run', '--observe=0'],
+      ddsServeLocalScriptPath,
+      if (runApp)
+        // When running DevTools via `flutter run`, the [flutterRunProcess]
+        // below will launch DevTools in the browser.
+        '--no-launch-browser'
+      else
+        // Only pass a build location if the server is serving the web assets
+        // (i.e. not when DevTools app is ran via `flutter run`).
+        '--devtools-build=$devToolsBuildLocation',
+      // Pass any args that were provided to our script along. This allows IDEs
+      // to pass `--machine` (etc.) so that this script can behave the same as
+      // the "dart devtools" command for testing local DevTools/server changes.
+      ...remainingArguments,
+    ], sdkOverride: serveWithDartSdk);
+    if (forMachine) {
+      // If --machine flag is true, then the output is a tool-readable JSON.
+      // Therefore, skip reading the process output and instead just run the
+      // process.
+      return processManager.runProcess(
```

Comment:
if we run the process I think we will wait for this process to exit, which will cause this script to fail for the case where `runApp` is true.

Reply:
Chatted offline that for now we can check whether --machine and --run-app are both set and if so throw an exception.